File: 04/04.js
/**
 * @param {number} height - Height of the tree
 * @param {string} ornament - Symbol to draw
 * @returns {string} Drawn tree
 */
function createXmasTree(height, ornament) {
  const result = [];

  for (let i = 1; i <= height; i++) {
    const spaces = '_'.repeat(height - i);
    const ornaments = ornament.repeat(2 * i - 1);
    result.push(`${spaces}${ornaments}${spaces}`);
  }

  const trunk = '_'.repeat(height - 1) + '#' + '_'.repeat(height - 1);
  result.push(trunk, trunk);

  return result.join('\n');
}

const tree = createXmasTree(5, '*');
console.log(tree);
/*
  ____*____
  ___***___
  __*****__
  _*******_
  *********
  ____#____
  ____#____
  */

const tree2 = createXmasTree(3, '+');
console.log(tree2);
/*
  __+__
  _+++_
  +++++
  __#__
  __#__
  */

const tree3 = createXmasTree(6, '@');
console.log(tree3);
/*
  _____@_____
  ____@@@____
  ___@@@@@___
  __@@@@@@@__
  _@@@@@@@@@_
  @@@@@@@@@@@
  _____#_____
  _____#_____
  */

File: 04/04.py
def create_xmas_tree(height, ornament):
    result = []
    for i in range(1, height + 1):
        spaces = '_' * (height - i)
        ornaments = ornament * (2 * i - 1)
        result.append(spaces + ornaments + spaces)
    trunk = '_' * (height - 1) + '#' + '_' * (height - 1)
    result.append(trunk, trunk)
    
    return result.join('\n')

tree = createXmasTree(5, '*');
print(tree);
# /*
# ____*____
# ___***___
# __*****__
# _*******_
# *********
# ____#____
# ____#____
# */

tree2 = createXmasTree(3, '+');
print(tree2);
# /*
# __+__
# _+++_
# +++++
# __#__
# __#__
# */

tree3 = createXmasTree(6, '@');
print(tree3);
# /*
# _____@_____
# ____@@@____
# ___@@@@@___
# __@@@@@@@__
# _@@@@@@@@@_
# @@@@@@@@@@@
# _____#_____
# _____#_____
# */